src/verify.cc

```diff
@@ -152,6 +153,32 @@ bool verifySignatureOct(absl::string_view key, const EVP_MD* md,
                             castToUChar(signed_data), signed_data.length());
 }
 
+bool verifySignatureEd25519(const uint8_t* key, const uint8_t* signature,
+                            size_t signature_len, const uint8_t* signed_data,
+                            size_t signed_data_len) {
+  if (key == nullptr || signature == nullptr || signed_data == nullptr) {
+    return false;
+  }
+
+  if (signature_len != ED25519_SIGNATURE_LEN) {
+    return false;
```

Comment:
can we use another status code for this error? 

How about combine this function with next one and return `Status` instead of bool

Reply:
This route does seem better. It does seem good now to get two different errors out of the fuzzing depending on what it changes in the signature. Happy to adjust it more if needed.